azurerm/internal/services/compute/validation.go

```diff
@@ -76,3 +76,18 @@ func validateName(maxLength int) func(i interface{}, k string) (warnings []strin
 		return
 	}
 }
+
+func ValidateDiskEncryptionSetName(i interface{}, k string) (warnings []string, errors []error) {
+	v, ok := i.(string)
+	if !ok {
+		errors = append(errors, fmt.Errorf("expected type of %s to be string", k))
+		return
+	}
+
+	// Supported characters for the name are a-z, A-Z, 0-9 and _. The maximum name length is 80 characters.
+	// despite the swagger says the constraint above, but a name that contains hyphens will also pass.
```

Comment:
can we confirm this validation with the portal, which is (unfortunately) the most reliable source of this information

Reply:
I have got the replies from the service team, and changed the regex.